concordium-std/src/types.rs

```diff
@@ -1018,27 +1019,172 @@ pub struct ExternCryptoPrimitives;
 #[repr(transparent)]
 pub struct PublicKeyEd25519(pub [u8; 32]);
 
+impl fmt::Display for PublicKeyEd25519 {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        for b in self.0 {
+            write!(f, "{:02x}", b)?;
+        }
+        Ok(())
+    }
+}
+
+impl FromStr for PublicKeyEd25519 {
+    type Err = ParseError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        if s.len() != 64 {
+            return Err(ParseError {});
+        }
+
+        let public_key: [u8; 32] = (0..s.len())
+            .step_by(2)
+            .map(|i| u8::from_str_radix(&s[i..i + 2], 16).unwrap())
+            .collect::<Vec<u8>>()
+            .as_slice()
+            .try_into()
+            .unwrap();
```

Comment:
```suggestion
        let mut public_key: [u8; 32] = [0u8; 32];
        for (i, place) in public_key.iter_mut().enumerate() {
            *place = u8:;from_str_radix(&s[2*i, 2*i+2], 16).map_err(|_| ParseError{})?;
        }
```

We should not use `unwrap` since parsing is supposed to be able to fail, and if it does so it should do it gracefully.